### ISSUE TITLE
AutoFDO Optimization

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/distribution#2711]
Try to start research on post-link optimizations
http://perl11.org/blog/bolt.html

**Comment by benbullard79** [comment]
Can we close thes report or make active?

**Comment by wilsonephillips** [comment]
There is no website active for this issue. What do we research? After 7 years the world is still turning without it. Has this been implemented in a different way or a later revision?

**Comment by rudiservo** [comment]
Updated to https://perl11.github.io/blog/bolt.html

**Comment by rudiservo** [comment]
Also adding this here.

https://docs.kernel.org/dev-tools/autofdo.html

**Comment by berolinux** [comment]
Largely blocked by bolt and the debuginfo generator hating each other - this may change in a future version.

**Comment by rudiservo** [comment]
@berolinux I think autofdo is a seperate thing from bolt, CachyOS is already using it since november 2024

https://cachyos.org/blog/2411-kernel-autofdo/

This is the issue and PR on cachy.
https://github.com/CachyOS/linux-cachyos/issues/325

I think we should separate Bolt from AutoFDO in two separate issues.

Synthetic benchmark performance gains where maybe 1% to 3%.

**Comment by benbullard79** [comment]
So close this or leave open?

**Comment by rudiservo** [comment]
Leave this with AUTOFDO

create a new one with BOLT

IMO this is not critical for UM.
A Nice to have
